os/arch/arm/src/common/up_checkstack.c

```diff
@@ -178,12 +178,12 @@ static size_t do_stackcheck(uintptr_t alloc, size_t size)
 
 size_t up_check_tcbstack(FAR struct tcb_s *tcb)
 {
-	return do_stackcheck((uintptr_t)tcb->stack_alloc_ptr, tcb->adj_stack_size);
+	return do_stackcheck((uintptr_t)tcb->stack_alloc_ptr, tcb->adj_stack_size + CONFIG_MPU_STACK_GUARD_SIZE);
```

Comment:
Because the purpose of this function is returning the size of remain stack, this should be as below
```
return do_stackcheck((uintptr_t)tcb->stack_alloc_ptr + CONFIG_MPU_STACK_GUARD_SIZE, tcb->adj_stack_size);
```